iceoryx2-ffi/cxx/include/iox2/response_mut_uninit.hpp

```diff
@@ -50,66 +55,122 @@ class ResponseMutUninit {
     auto user_header_mut() -> T&;
 
     /// Returns a reference to the payload of the response.
-    auto payload() const -> const ResponsePayload&;
+    template <typename T = ResponsePayload, typename = std::enable_if_t<!iox::IsSlice<T>::VALUE, void>>
+    auto payload() const -> const T&;
+
+    /// Returns a reference to the payload of the response.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<iox::IsSlice<T>::VALUE, void>>
+    auto payload() const -> iox::ImmutableSlice<ValueType>;
 
     /// Returns a mutable reference to the payload of the response.
-    auto payload_mut() -> ResponsePayload&;
+    template <typename T = ResponsePayload, typename = std::enable_if_t<!iox::IsSlice<T>::VALUE, void>>
+    auto payload_mut() -> T&;
+
+    /// Returns a mutable reference to the payload of the response.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<iox::IsSlice<T>::VALUE, void>>
+    auto payload_mut() -> iox::MutableSlice<ValueType>;
+
+    /// Writes the provided payload into the [`ResponseMutUninit`] and returns an initialized
+    /// [`ResponseMut`] that is ready to be sent.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<!iox::IsSlice<T>::VALUE, T>>
+    void write_payload(ResponsePayload&& payload);
+
+    /// Writes the provided payload into the [`ResponseMutUninit`] and returns an initialized
+    /// [`ResponseMut`] that is ready to be sent.
+    template <typename T = ResponsePayload, typename = std::enable_if_t<iox::IsSlice<T>::VALUE, T>>
+    void write_from_slice(iox::ImmutableSlice<ValueType>& value);
```

Comment:
According to the documentation and the rust counterpart it should return an initialized `ResponseMut`.